code/datums/components/links_to_operating_computers.dm

```diff
@@ -0,0 +1,126 @@
+/// Anything with this component will show surgeries performed on them
+/// on the operating computer, and optionally provide upgraded surgeries.
+/datum/component/links_to_operating_computers
+	/// Should this object allow the upgraded surgeries?
+	var/provide_upgraded_surgeries = FALSE
+
+	VAR_PRIVATE
+		/// Current known patients, in order of least to most recently added
+		list/patients = list()
+
+/datum/component/links_to_operating_computers/Initialize(provide_upgraded_surgeries)
+	. = ..()
+
+	if (!ismovable(parent))
+		stack_trace("Parent must be a movable")
+		return COMPONENT_INCOMPATIBLE
+
+	if (!isnull(provide_upgraded_surgeries))
+		src.provide_upgraded_surgeries = provide_upgraded_surgeries
+
+/datum/component/links_to_operating_computers/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(on_movable_moved))
+	register_to_loc()
+
+/datum/component/links_to_operating_computers/UnregisterFromParent()
+	var/atom/atom_parent = parent
+
+	UnregisterSignal(atom_parent, COMSIG_MOVABLE_MOVED)
+	unregister_loc(atom_parent.loc)
+
+	for (var/patient in patients)
+		SEND_SIGNAL(src, COMSIG_LINKS_TO_OPERATING_COMPUTERS_PATIENT_REMOVED, patient)
+
+	patients.Cut()
+
+/datum/component/links_to_operating_computers/proc/on_atom_entered(source, mob/living/carbon/enterer)
+	SIGNAL_HANDLER
+
+	if (enterer in patients)
+		return
+
+	check_atom_on_object(enterer)
+
+/datum/component/links_to_operating_computers/proc/check_atom_on_object(atom)
+	if (!iscarbon(atom))
+		return
+
+	RegisterSignal(atom, COMSIG_LIVING_SET_BODY_POSITION, PROC_REF(on_enterer_body_position_changed))
+	check_patient(atom)
+
+/datum/component/links_to_operating_computers/proc/on_atom_exited(source, mob/living/carbon/exiter)
+	SIGNAL_HANDLER
+
+	if (!istype(exiter))
+		return
```

Comment:
why istype here and iscarbon one above?

Reply:
Artifact of some previous code, will change